src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java

```diff
@@ -0,0 +1,113 @@
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.service.CategoryServiceImp;
+import mate.academy.springboot.datajpa.service.ProductServiceImp;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductServiceImp productService;
+    private final CategoryServiceImp categoryService;
+    private final ProductMapper mapper;
+
+    @PostConstruct
+    public void init() {
+        Category meat = new Category().setName("meat");
+        Category milk = new Category().setName("milk");
+        Category bread = new Category().setName("bread");
+        categoryService.create(meat);
+        categoryService.create(milk);
+        categoryService.create(bread);
+        Product firstMeat = new Product().setCategory(meat).setPrice(15).setTitle("Fresh");
+        Product secondMeat = new Product().setCategory(meat).setPrice(25).setTitle("Middle");
+        Product firstMilk = new Product().setCategory(milk).setPrice(40).setTitle("Prostokvash");
+        Product secondMilk = new Product().setCategory(milk).setPrice(50).setTitle("Yagotin");
+        Product firstBread = new Product().setCategory(bread).setPrice(15).setTitle("Rudnia");
+        productService.create(firstMeat);
+        productService.create(secondMeat);
+        productService.create(firstMilk);
+        productService.create(secondMilk);
+        productService.create(firstBread);
+    }
+
+    @PostMapping
+    public ResponseEntity<ProductResponse> create(@Valid @RequestBody ProductRequest request) {
+        String categoryName = request.getCategoryName();
+        Category category = categoryService.findByName(categoryName);
+        Product newProduct = mapper.mapToEntity(request);
+        newProduct.setCategory(category);
+        Product product = productService.create(newProduct);
+        ProductResponse response = mapper.mapToDto(product);
+        return new ResponseEntity<>(response, HttpStatus.OK);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<ProductResponse> get(@PathVariable Long id) {
+        Product product = productService.getById(id);
+        ProductResponse response = product == null ? null : mapper.mapToDto(product);
```

Comment:
don't need to make such checks in controller, return from repository method Optional and in case if optional empty throw an exception on service layer